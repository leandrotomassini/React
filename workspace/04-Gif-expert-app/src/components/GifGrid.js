import React, { useState, useEffect } from 'react'

export const GifGrid = ({ category }) => {

    const [images, setimages] = useState([]);

    useEffect(() => {
        getGifs();
    }, []);

    const getGifs = async () => {

        const url = 'https://api.giphy.com/v1/gifs/search?q=Rick+and+Morty&limit=10&api_key=fwHVLO449TE0bty2sFoKDrDRETR2g4F3';
        const resp = await fetch(url);
        const { data } = await resp.json();

        const gifs = data.map(img => {
            return {
                id: img.id,
                title: img.title,
                url: img.images?.downsized_medium.url
            }
        });

        console.log(gifs);
        setimages(gifs);
    }

    getGifs();

    return (
        <div>
            <h3>{category}</h3>
            <ol>
                {
                    images.map(({ id, title }) => (
                        <li key={id}>{title}</li>
                    ))
                }
            </ol>
        </div>
    )
}
