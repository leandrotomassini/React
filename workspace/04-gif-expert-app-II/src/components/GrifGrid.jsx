import { useFetchGifs } from '../hooks/useFetchGifs';

import { GridItem } from './GridItem';


export const GrifGrid = ({ category }) => {

    const { images, isLoading } = useFetchGifs(category);


    return (
        <>
            <h3>{category}</h3>

            {
                isLoading && (<p>Cargando...</p>)
            }

            <div className='card-grid'>
                    {
                        images.map((image) => (
                            <GridItem 
                                key={image.id}
                                {...image}
                            />
                        ))
                    }
            </div>
        </>
    );
}
