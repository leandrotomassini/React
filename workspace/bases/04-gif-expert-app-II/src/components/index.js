export * from './AddCategory';
export * from './GridItem';
export * from './GrifGrid';