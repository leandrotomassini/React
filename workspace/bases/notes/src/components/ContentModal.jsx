

function ContentModal( { title, body, idx } ) {
  return (
    <div className="fade modal modal-xl" id={ "note" + idx } data-bs-backdrop="static" data-bs-keyboard="false" tabIndex="-1" aria-labelledby={ "note" + idx + "Label" } aria-hidden="true">
      <div className="modal- modal-dialog modal-dialog-scrollable modal-fullscreen-xl-down">
        <div className="modal-content">
          <div className="modal-header">
            <h1 className="fs-5 modal-title" id={ "note" + idx + "Label" }>Content modal</h1>
            <button type="button" className="btn-close" data-bs-dismiss="modal" aria-label="Close"></button>
          </div>
          <div className="modal-body">
            <h2>{ title }</h2>
            <p>{ body }</p>
          </div>
          <div className="modal-footer">
            <button type="button" className="btn btn-secondary" data-bs-dismiss="modal">
              <span className="pe-2">Close</span>
              <i className="bi bi-fullscreen-exit"></i>
            </button>
            <button type="button" className="btn btn-primary" data-bs-target="#editormodal" data-bs-toggle="modal">
              <span className="pe-2">Edit</span>
              <i className="bi bi-pencil"></i>
            </button>
          </div>
        </div>
      </div>
    </div>
  );
}

export default ContentModal;