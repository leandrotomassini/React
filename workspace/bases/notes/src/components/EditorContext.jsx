import { createContext, useRef } from 'react';
import EditorJS from '@editorjs/editorjs';

export const EditorContext = createContext();

function EditorContextProvider( props ) {

  const editorInstanceRef = useRef();

  const initEditor = () => {

    const editor = new EditorJS( {
      holder: "editorjs",
      placeholder: "Let's take a note!",
      tools: {}
    } );

    editorInstanceRef.current = editor;
  };

  return (
    <EditorContextProvider value={ { initEditor, editorInstanceRef } }>
      { props.children }
    </EditorContextProvider>
  );
}

export default EditorContextProvider;