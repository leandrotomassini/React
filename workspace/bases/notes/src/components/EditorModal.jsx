

function EditorModal() {

  return (
    <div className="fade modal modal-lg" id="editormodal" data-bs-backdrop="static" data-bs-keyboard="false" tabIndex="-1" aria-labelledby="editormodalLabel" aria-hidden="true">
      <div className="modal- modal-dialog modal-dialog-centered modal-dialog-scrollable modal-fullscreen-lg-down">
        <div className="modal-content">
          <div className="modal-header">
            <h1 className="fs-5 modal-title" id="editormodalLabel">Editor modal</h1>
            <button type="button" className="btn-close" data-bs-dismiss="modal" aria-label="Close"></button>
          </div>
          <div className="modal-body">
            Modal body
          </div>
          <div className="modal-footer">
            <button type="button" className="btn btn-secondary" data-bs-dismiss="modal">Close</button>
            <button type="button" className="btn btn-primary">Save Note</button>
          </div>
        </div>
      </div>
    </div>
  );
}

export default EditorModal;