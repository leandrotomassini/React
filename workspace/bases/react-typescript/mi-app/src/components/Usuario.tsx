import { useState } from 'react';


interface IUsuario {
    uid: string;
    name: string;
}


export const Usuario = () => {

    const [user, setuser] = useState<IUsuario>();

    const login = () => {
        setuser({
            uid: 'ABC123',
            name: 'Leandro Tomassini'
        })
    }


    return (
        <div className='mt-5'>

            <h3>Usuario: useState</h3>

            <button className='btn btn-outline-primary' onClick={login}>
                Login
            </button>
            {
                (!user)
                    ? <pre>No hay usuario</pre>
                    : <pre>{JSON.stringify(user)}</pre>
            }



        </div>
    );
}
