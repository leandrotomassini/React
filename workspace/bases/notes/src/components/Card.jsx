import ContentModal from './ContentModal';


function Card( { title, body, idx } ) {

  return (
    <div className="mx-2 mb-3">
      <div className="card">
        <div className="z-2 ms-auto">
          <button className="d-flex align-items-center mt-2 btn btn-outline-danger btn-sm me-3">
            <span className="pe-1">Delete</span>
            <i className="bi bi-trash3 fs-6"></i>
          </button>
        </div>
        <div className="card-body">
          <h2>{ title }</h2>
          <p>{ body }</p>
          <a type="button" className="stretched-link" data-bs-toggle="modal" data-bs-target={ "#note" + idx }></a>
        </div>
      </div>
      <ContentModal title={ title } body={ body } id={ idx } />
    </div>
  );
}

export default Card;