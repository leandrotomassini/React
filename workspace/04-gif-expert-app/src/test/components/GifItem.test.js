import { render } from '@testing-library/react';

import { GifItem } from '../../components/GifItem';


describe('Pruebas en <GifItem />', () => {

    const title = 'Saitama';
    const url = 'https://www.google.com'


    test('Debe de hacer match con el snapshop', () => {
        const { container } = render(<GifItem title={title} url={url} />)
        expect(container).toMatchSnapshot();
    });

    

});