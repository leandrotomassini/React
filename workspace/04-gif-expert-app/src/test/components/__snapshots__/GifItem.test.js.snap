// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <GifItem /> Debe de hacer match con el snapshop 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="Saitama"
      src="https://www.google.com"
    />
    <p>
      Saitama
    </p>
  </div>
</div>
`;
